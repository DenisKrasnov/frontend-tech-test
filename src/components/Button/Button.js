/* eslint-disable react/button-has-type */
// 'type' is generic and comes from props
import React from "react";
import { string, func, bool } from "prop-types";

import Spinner from "../Spinner";
import styles from "./Button.scss";

const Button = props => {
  const { children, loading, onClick, type } = props;

  return (
    <button
      className={styles.Button}
      disabled={loading}
      onClick={onClick}
      type={type}
    >
      {loading ? <Spinner /> : children}
    </button>
  );
};

Button.defaultProps = {
  loading: false,
  type: "submit",
  onClick: () => {},
};

Button.propTypes = {
  children: string.isRequired,
  onClick: func,
  loading: bool,
  type: string,
};

export default Button;
