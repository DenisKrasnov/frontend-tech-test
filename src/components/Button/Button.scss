.button {
  display: block;
  padding: 10px 30px;
  margin: 20px auto 0 auto;
  border: 0;
  border-radius: 4px;
  outline: 0;
  background-color: $button-primary-color;
  font-weight: 600;
  line-height: 20px;
  color: #fff;
  user-select: none;
  overflow: hidden;

  &:hover {
    background: $button-hover-color;
  }
}
