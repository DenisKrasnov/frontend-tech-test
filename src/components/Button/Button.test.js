import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  it("renders", () => {
    const component = shallow(<Button>Name</Button>);
    expect(component).toMatchSnapshot();
  });

  it("renders in a loading mode", () => {
    const component = shallow(<Button loading={true}>Name</Button>);
    expect(component).toMatchSnapshot();
  });

  it("triggers onClick", () => {
    const onClickSpy = jest.fn();
    const component = shallow(<Button onClick={onClickSpy}>Name</Button>);

    component.find("button.Button").simulate("click");
    expect(onClickSpy).toHaveBeenCalled();
  });
});
