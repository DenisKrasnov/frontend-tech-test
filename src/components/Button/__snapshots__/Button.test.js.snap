// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  Name
</button>
`;

exports[`Button renders in a loading mode 1`] = `
<button
  className="Button"
  disabled={true}
  onClick={[Function]}
  type="submit"
>
  <Spinner />
</button>
`;
